### PR TITLE
Added Application Architecture

### PR DESCRIPTION
**Pull request by Hossam-Elgmmal**
## Description

Added Architecture for the app **By Feature**.
Every feature has domain / di / data / presentation (as needed).
Made basic setup for each feature to guarantee consistency throughout the app.

Updated Kotlin version to 2.1.0
Increased **targetSdk** to **35** so it is the same as **compileSdk**
Fixed bug "failed to make module as instrumented mockk produces duplicate files"
Downgraded hilt to 2.52 because version 2.53 has an issue with kapt

---

### setup features:
* Event
* Onboarding
* Map
* Settings
* User
* Room
* Retrofit
* Datastore

Added the screens relative to features.

---

## Screenshots (if applicable)

> Not applicable
---

## Checklist
Please ensure the following tasks are completed:

- [ ] Unit tests are included for the new functionality (if applicable).
- [x] Code has been properly documented.
- [x] Changes have been tested and verified.
---

## Additional Comments

> Test as you go

### CODE REVIEW

**Review by Zeinab979 — APPROVED** [review]

File: app/src/main/java/com/gdgevents/gdgeventsapp/core/datastore/di/UserPreferencesModule.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import dagger.Module
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+
+@Module
+@InstallIn(SingletonComponent::class)
+abstract class UserPreferencesRepoModule {
+
+    @Binds
+    abstract fun bindsMapRepo(
```
Comment (qamarelsafadi) [inline app/src/main/java/com/gdgevents/gdgeventsapp/core/datastore/di/UserPreferencesModule.kt:15]: bindMapRepo 👀?

File: app/src/main/java/com/gdgevents/gdgeventsapp/core/retrofit/data/NetworkRepoImpl.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.gdgevents.gdgeventsapp.core.retrofit.data
```
Comment (qamarelsafadi) [inline app/src/main/java/com/gdgevents/gdgeventsapp/core/retrofit/data/NetworkRepoImpl.kt:1]: Instead retrofit as package we can replace it with remote

File: app/src/main/java/com/gdgevents/gdgeventsapp/core/work/data/SyncWorker.kt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+
+@HiltWorker
+class SyncWorker @AssistedInject constructor(
```
Comment (qamarelsafadi) [inline app/src/main/java/com/gdgevents/gdgeventsapp/core/work/data/SyncWorker.kt:11]: Lets remove the work manager in this phase, just to not to increase complexity

**Review by qamarelsafadi — APPROVED** [review]

**Comment by qamarelsafadi** [comment]
@Hossam-Elgmmal let's wait until the people who are assigned for this issue to make their review, @alaafawzyy @ahmedgamalz

**Pull request merged**